src/components/PostDetails.tsx

```diff
@@ -1,107 +1,123 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { client, deleteComment } from '../utils/fetchClient';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [isWritingComment, setIsWritingComment] = useState(false);
+
+  const loadComments = async (postId: number) => {
+    setIsWritingComment(false);
+    setIsLoading(true);
+    setError('');
+
+    try {
+      const fetchedComments = await client.get<Comment[]>(
+        `/comments?postId=${postId}`,
+      );
+
+      setComments(fetchedComments);
+    } catch (err) {
+      setError('Something went wrong');
```

Comment:
The error message 'Something went wrong' is quite generic. Consider providing more specific error messages based on the type of error encountered to improve user feedback and debugging.